internal/server/server.go

```diff
@@ -108,19 +198,92 @@ func (s *Server) handleLogin() http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
 		session := s.getSession(r)
 
-		rs, err := random.SecureString(32)
+		randomStr, err := random.SecureString(32)
```

Comment:
I think we're actually missing a state check here.

According to https://pkg.go.dev/golang.org/x/oauth2#Config.AuthCodeURL we should be storing this state value in the session and then validating it against the state returned as a query param in the callback handler.

Reply:
I think that is only applicable if our redirect URL goes to our callback. I think purpose there is to make sure state doesn't change for security.
In our case, we hit /login, copy the code. Go back to the root page and then manually hit /callback. So there's no way for the URL to pass a state.

Reply:
I can make a comment in the code, probably same question will appear in future.

Reply:
The login flow always calls the callback handler after hitting dex.  The callback handler is where state should be evaluated similarly to the `nonce`.   Its seems like this is a security feature we missed in our initial implementation of the auth portal.

Reply:
I spoke with @yuzhouliu9 via slack.  The RedirectURL is out-of-band and acts like a SPA, which means the OIDC provider does not redirect back to /callback of the auth portal.  The user is activating this by clicking on the form button on the index.html page.

I do feel there is an opportunity to still protect against potential CSRF attacks by adding a CSRF token as a hidden form element and adjusting the form method to POST.   Given that this is a new feature I think it is best to implement in a separate PR where that change can be discussed separately.